diff --git a/node_modules/@types/invity-api/index.d.ts b/node_modules/@types/invity-api/index.d.ts
index 0c62fb2..038f3a1 100755
--- a/node_modules/@types/invity-api/index.d.ts
+++ b/node_modules/@types/invity-api/index.d.ts
@@ -157,6 +157,8 @@ export type ExchangeTradeStatus =
     | 'SENDING' // send tx was created, waiting for send tx to be sent
     | 'CONFIRMING' // send tx was sent, waiting for tx to be mined (not used for Trezor Wallet)
     | 'CONVERTING' // send tx was mined, money is on exchange, receive tx not yet created
+    | 'APPROVAL_REQ' // it is necessary to perform APPROVAL transaction for DEX
+    | 'APPROVAL_PENDING' // waiting for DEX approval tx to be confirmed
     | ExchangeTradeFinalStatus;
 
 export type ExchangeFee =
@@ -176,6 +178,7 @@ export interface ExchangeProviderInfo {
     logo: string; // changenow-icon.jpg
     isActive: boolean;
     isFixedRate: boolean;
+    isDex: boolean;
     buyTickers: string[];
     sellTickers: string[];
     addressFormats: StringMap; // specification of formats required by selected exchange
@@ -198,6 +201,12 @@ export interface ExchangeCoinInfo {
 
 export type ExchangeCoinListResponse = ExchangeCoinInfo[];
 
+export type DexApprovalType =
+    | 'MINIMAL' // MINIMAL (default) is the lowest necessary to swap sendStringAmount
+    | 'INFINITE' // approves infinite amount
+    | 'ZERO' // resets approval
+    | 'PRESET'; // PRESET takes value from approvalStringAmount
+
 export interface ExchangeTrade {
     send?: string; // BTC
     sendStringAmount?: string; // "0.01"
@@ -225,6 +234,20 @@ export interface ExchangeTrade {
     extraFieldDescription?: CoinExtraField;
     tags?: ExchangeTradeTag[];
     id?: string; // internal DB id
+    // DEX extra fields
+    isDex?: boolean;
+    approvalGasEstimate?: number; // gas estimate of the approval transaction
+    approvalType?: DexApprovalType;
+    approvalStringAmount?: string; // amount to approve in case of PRESET or to return the minimal necessary amount in case of MINIMAL
+    approvalSendTxHash?: string; // txid of approval transaction
+    swapGasEstimate?: number; // gas estimate of the swap transaction
+    dexTx?: {
+        // tx data for approval or swap transaction
+        from: string;
+        to: string;
+        data: string;
+        value: string;
+    };
     // locally used fields
     offerType?: 'bestRate' | 'favorite';
 }
@@ -245,6 +268,7 @@ export interface ExchangeTradeQuoteRequest {
     send: string; // BTC
     receive: string; // LTC
     sendStringAmount?: string; // "0.01"
+    dex?: 'enable' | 'exclusively'; // 'enable' means add dex offers, 'exclusively' means only dex offers
 }
 
 export type ExchangeTradeQuoteResponse = ExchangeTrade[];
